apps/roam/src/components/LeftSidebarView.tsx

```diff
@@ -407,20 +410,109 @@ const LeftSidebarView = ({ onloadArgs }: { onloadArgs: OnloadArgs }) => {
+
+  const config = getFormattedConfigTree().leftSidebar;
+
+  if ((config.favoritesMigrated as BooleanSetting).value) return;
+
+  let leftSidebarUid = config.uid;
+  if (leftSidebarUid) {
+    const leftSidebarTree = getBasicTreeByParentUid(leftSidebarUid);
+    const hasAnyPersonalSection = leftSidebarTree.some((node) =>
+      node.text.endsWith("/Personal-Section"),
+    );
+    if (hasAnyPersonalSection) {
+      await createBlock({
+        parentUid: leftSidebarUid,
+        node: { text: "Favorites Migrated" },
+      });
+      refreshConfigTree();
+      return;
+    }
+  }
+
+  const results = window.roamAlphaAPI.q(`
+    [:find ?title 
+     :where [?e :page/sidebar]
+            [?e :node/title ?title]]
+  `);
+  const titles = (results as string[][]).map(([title]) => title);
+
+  if (!leftSidebarUid) {
+    const tree = getBasicTreeByParentUid(configPageUid);
+    const found = tree.find((n) => n.text === "Left Sidebar");
+    if (found) {
+      leftSidebarUid = found.uid;
+    } else {
+      leftSidebarUid = await createBlock({
+        parentUid: configPageUid,
+        node: { text: "Left Sidebar" },
+      });
+    }
+  }
+
+  let globalSectionUid = config.global.uid;
+  if (!globalSectionUid) {
+    const tree = getBasicTreeByParentUid(leftSidebarUid);
+    const found = tree.find((n) => n.text === "Global-Section");
+    if (found) {
+      globalSectionUid = found.uid;
+    } else {
+      globalSectionUid = await createBlock({
+        parentUid: leftSidebarUid,
+        node: { text: "Global-Section" },
+      });
+    }
+  }
+
+  let childrenUid = config.global.childrenUid;
+  if (!childrenUid) {
+    const tree = getBasicTreeByParentUid(globalSectionUid);
+    const found = tree.find((n) => n.text === "Children");
+    if (found) {
+      childrenUid = found.uid;
+    } else {
+      childrenUid = await createBlock({
+        parentUid: globalSectionUid,
+        node: { text: "Children" },
+      });
+    }
+  }
+
+  const childrenTree = getBasicTreeByParentUid(childrenUid);
+  const existingTitles = new Set(childrenTree.map((c) => c.text));
+  const newTitles = titles.filter((t) => !existingTitles.has(t));
+
+  if (newTitles.length > 0) {
+    await Promise.all(
+      newTitles.map((text) =>
+        createBlock({ parentUid: childrenUid, node: { text } }),
+      ),
+    );
+    refreshAndNotify();
+  }
+
+  await createBlock({
+    parentUid: leftSidebarUid,
+    node: { text: "Favorites Migrated" },
+  });
+  refreshConfigTree();
+};
+
+export const mountLeftSidebar = async (
   wrapper: HTMLElement,
   onloadArgs: OnloadArgs,
-): void => {
+): Promise<void> => {
   if (!wrapper) return;
-  wrapper.innerHTML = "";
 
   const id = "dg-left-sidebar-root";
   let root = wrapper.querySelector(`#${id}`) as HTMLDivElement;
   if (!root) {
-    const existingStarred = wrapper.querySelector(".starred-pages");
-    if (existingStarred) {
-      existingStarred.remove();
-    }
+    await migrateFavorites();
```

Comment:
Ideally we only run this when the left sidebar flag is turned `on`.  It is probably unncessary to run every load.

Reply:
yeah and this will only run when the flag is turned on

Reply:
The point is that running both `getPageUidByPageTitle()` and `getFormattedConfigTree()` on every load is unnecessary.  `migrateFavorites()` should only be run once when a user enables the sidebar and we migrate the favorites.  There are multiple API calls here that could be avoided.